### PR TITLE
Feature/dbkvs tests

### PR DESCRIPTION
**Pull request by jboreiko**
Dbkvs testing + functionality modifications to get these tests passing.

### CODE REVIEW

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java [hunk 1]
```diff
@@ -795,7 +829,17 @@ public Void apply(DbDdlTable table) {
     private <T> T runDdl(TableReference tableRef, Function<DbDdlTable, T> runner) {
         ConnectionSupplier conns = new ConnectionSupplier(connections);
         try {
-            return runner.apply(dbTables.createDdl(tableRef.getQualifiedName(), conns));
+            /* The ddl actions can used both the fully qualified name and the internal name */
+            return runner.apply(dbTables.createDdl(tableRef, conns));
+        } finally {
+            conns.close();
```
Comment (schlosna) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java]: probably worth a separate PR, but can we use try-with-resources for these to get proper suppressed exceptions?

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/OracleDbTableFactory.java [hunk 2]
```diff
@@ -44,11 +45,27 @@ public DbMetadataTable createMetadata(String tableName, ConnectionSupplier conns
...
+        /* No initialization required for Oracle */
+        return new DbTableInitializer() {
+            @Override
+            public void createUtilityTables() {
+                //no op
+            }
+
+            @Override
+            public void createMetadataTable(String metadataTableName) {
+                //no op - will be required for Oracle to work
```
Comment (schlosna) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/OracleDbTableFactory.java]: Do we define the schemas in SQL somewhere to pre-create these tables?

Comment (schlosna) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/OracleDbTableFactory.java]: Should probably do basic verification check that utility & metadata table exists.

**Comment by joelea** [comment]
Can't repro locally as I get stymied by a different error:

`Caused by: com.palantir.util.sql.VerboseSQLException: While attempting to execute the SQL '/* UnregisteredSQLString */ ( /* SIMPLE_ROW_SELECT_ONE_TEMPLATE_PSQL (ns.table1) */  SELECT /*+ INDEX(t pk_a_ns.table1) */    DISTINCT row_name, ? as batch_num  FROM a_ns.table1 t  WHERE  t.row_name >= ?   ORDER BY row_name ASC LIMIT 1)' we caught a SQLException.
The SQL was executed with the following bind args:
(java.lang.Integer: '0', [B: '[B@340b9973')
End of verbose SQLException error message: ERROR: relation "a_ns.table1" does not exist
  Position: 168
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2101)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1834)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:255)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:510)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:386)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery(AbstractJdbc2Statement.java:271)
	at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.palantir.nexus.db.pool.InterceptorStatement.handleInvocation(InterceptorStatement.java:52)
	at com.google.common.reflect.AbstractInvocationHandler.invoke(AbstractInvocationHandler.java:87)
	at com.sun.proxy.$Proxy19.executeQuery(Unknown Source)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
	at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.palantir.nexus.db.sql.BasicSQL$BlobCleanupPreparedStatement.invoke(BasicSQL.java:344)
	at com.sun.proxy.$Proxy20.executeQuery(Unknown Source)
	at com.palantir.nexus.db.sql.BasicSQL$2.call(BasicSQL.java:425)
	at com.palantir.nexus.db.sql.BasicSQL$2.call(BasicSQL.java:416)
	at com.palantir.common.concurrent.ThreadNamingCallable.call(ThreadNamingCallable.java:55)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.palantir.common.concurrent.ForwardingRunnableFuture.run(ForwardingRunnableFuture.java:55)
	at com.palantir.common.concurrent.PTExecutors$5.run(PTExecutors.java:486)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)`

**Comment by rjullman** [comment]
@joelea that error indicates that somewhere in the code is not converting the TableReference to a useable String (there should not be dots (".") in table names).  This is an overarching problem with the way the code is setup since many different ddl classes have to access the metadata table which needs access to both a TableReference and a table's qualified name.  Refactoring the crosscutting nature of the metadata table should probably happen in another commit.

File: atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueService.java [hunk 3]
```diff
@@ -300,8 +300,7 @@ public Void call() throws Exception {
         }
     }
 
-
-    protected static String internalTableName(TableReference tableRef) {
+    public static String internalTableName(TableReference tableRef) {
```
Comment (rjullman) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueService.java]: This shouldn't be public.  I see that it has to be for this PR to work due to the crosscutting nature of the dbkvs metadata table.  I'm fine with this for now, but we should definitely try to refactor metadata table usage so that we can make the table reference to underlying table string kvs protected.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java [hunk 4]
```diff
@@ -96,7 +98,11 @@ public static DbKvs create(DbKeyValueServiceConfig config) {
         ReentrantManagedConnectionSupplier connSupplier = new ReentrantManagedConnectionSupplier(connManager);
         SqlConnectionSupplier sqlConnSupplier = getSimpleTimedSqlConnectionSupplier(connSupplier);
 
-        return new DbKvs(config, config.tableFactorySupplier().get(), sqlConnSupplier);
+        DbKvs dbKvs = new DbKvs(config, config.tableFactorySupplier().get(), sqlConnSupplier);
+
+        dbKvs.init();
+
+        return dbKvs;
```
Comment (rjullman) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java]: Internal user of dbkvs may not call DbKvs.create() and thus miss initialization code.  This shouldn't be a problem for the current initialization code here (in fact it happens to be somewhat of a feature in this case), but we need to make sure to remember this in the future (so internal uses don't end up borked in certain code paths).

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/OracleDbTableFactory.java (hunk 2 above)
Comment (jboreiko) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/OracleDbTableFactory.java]: We currently don't that will be included in another PR when the testing of Oracle is turned on.  At the moment we are not going to allow OSS to turn on Oracle.  For the verification check, do you mean beyond creating them if they don't exist as we do for Postgres or are you recommending an additional check.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java (hunk 4 above)
Comment (jboreiko) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java]: My hope is to in time deprecate the internal configuration code path entirely to avoid this issue.

**Comment by rjullman** [comment]
I'm fine with merging this when internal code associated with this passes CI.
